### ISSUE TITLE
mo-doc generation using referenced type name vs data structure

### ISSUE THREAD

**Issue by Gekctek** [dfinity/motoko#4290]
Im having an issue where I have a public facing part of my Motoko library and an internal facing section to have some code not be exposed, while keeping all of its functionality

So in many of my files I have `public type X = InternalTypes.X` for many different types, exposing the ones that i want exposed. This functionally works just fine but in the generated documentation i get 
```
## Type `X`
``` motoko no-repl
type X= InternalTypes.X
```
```
Which is super unhelpful. Is it possible to have the types never show the referenced name and only show the data structure itself?
Or are there any other options that you can think of?
I dont want to have to duplicate all the types across all my files just to get the documentation to work

**Comment by ggreif** [comment]
Good point I see a similar thing in connection with imported (and non-renamed) private type names. These get suffixed in the `.did` file with `__1` etc., and cause bloat and confusion only.

The example is
``` Motoko
// in main.mo
    type Credit = Types.Credit;
```
My suggestions:
- if there is no renaming, don't generate suffixes
- if the RHS is a public type and the LHS private, keep using the public one.

**Comment by rvanasa** [comment]
Relevant source code locations:
* [`declaration_header`](https://github.com/dfinity/motoko/blob/404af519fc028a1c5475df3719b16d7a2409aa16/src/docs/plain.ml#L250)
* [`plain_of_doc_typ`](https://github.com/dfinity/motoko/blob/404af519fc028a1c5475df3719b16d7a2409aa16/src/docs/plain.ml#L210-L213)
* [`extract_doc`](https://github.com/dfinity/motoko/blob/404af519fc028a1c5475df3719b16d7a2409aa16/src/docs/extract.ml#L155-L160)

Fixing this seems nontrivial; I encountered a similar issue when implementing type information tooltips in the VS Code extension. @ggreif @crusso, this relies on having some way to recover the original `Syntax.typ` from a `Syntax.PathT`.

**Comment by crusso** [comment]
Unfortunately, I think the current design of mo-doc is too simplistic to support that. It currently only has access to the syntax of Motoko code, no type information at all. To do better, we would need to replace mo-doc by a tool that is integrated with the type checker, perhaps by adding a special flag to also generate doc  to moc.

**Comment by Gekctek** [comment]
Any thoughts on alternatives besides type definition duplication?
